common/src/main/java/com/microsoft/identity/common/adal/internal/util/StringExtensions.java

```diff
@@ -247,4 +254,74 @@ public static boolean hasPrefixInHeader(final String value, final String prefix)
     public static String base64UrlEncodeToString(final String message) {
         return Base64.encodeToString(message.getBytes(Charset.forName(ENCODING_UTF8)), Base64.URL_SAFE | Base64.NO_WRAP);
     }
+
+
+
+    /**
+     * Decode the given url, and convert it into map with the given delimiter.
+     *
+     * @param url       The url to decode for.
+     * @param delimiter The delimiter used to parse the url string.
+     * @return The Map of the items decoded with the given delimiter.
+     */
+    public static Map<String, String> decodeUrlToMap(final String url, final String delimiter) {
```

Comment:
This method has a [known bug](https://github.com/AzureAD/microsoft-authentication-library-for-android/issues/253).

Reply:
Alternative approach to consider: https://stackoverflow.com/a/34040238/741827

Reply:
Given there are known bugs in this implementation liable to affect our B2C customers, I suggest we remove this implementation and favor using a tested library or Android platform implementation